go-client/generate.go

```diff
@@ -0,0 +1,58 @@
+//go:generate go run generate.go
```

Comment:
For (3), the reason I don't use raw command-line commands is because each command would be 450+ columns long (even when doing each file separately!), and Go does not offer a way to split `go:generate` comments across multiple lines. The reason I do the looping in the binary is because you have to provide module paths for each proto file _and its dependencies too_. So, if I did each file separately, I would have to pass in arguments for its dependencies as well, which greatly complicates things. Most other Go Projects do not have this issue because they modify their `.proto` files to be friendly to protoc-gen-go, so they don't have to specify all of the module paths.

Reply:
I noticed that some scripts explicitly build `proto-gen-go` and `proto-gen-go-grpc` as part of another `go:generate` and then use it to do the generation. Should we do this?

Reply:
Something like that is what I was trying to suggest in (5). Something like:

```go
//go:generate go build xyz
//go:generate xyz a b c
//go:generate xyz d e f
```